packages/core/src/components/Button/Button.tsx

```diff
@@ -116,15 +119,18 @@ export const HvButton = forwardRef<HTMLButtonElement, HvButtonProps>(
       }
     };
 
+    const StyledComponent = useMemo(() => StyledButton(component), [component]);
+
     return (
-      <StyledButton
+      <StyledComponent
         id={id}
         className={clsx(className, classes?.root, buttonClasses.root)}
         ref={ref}
         onClick={onClick}
         disabled={disabled}
         onFocus={onFocusHandler}
         onBlur={onBlurHandler}
+        tabindex="0" // Allow keyboard focus for all elements
```

Comment:
Do we really want this? If we do `component="div"` without an `onClick`, why should be `HvButton` be focusable? We'd be focusing an element that "does nothing".

Come to think of it, we probably shouldn't be promoting using `component="div"`, for accessibility reasons.

Reply:
I added this because inside the button component we use the `onFocus` and `onBlur` handlers to add a focus styling around the button. Without this the styling was not being applied to the button if another component was used.

Maybe we should change this and only allow the styling to be applied if the `onClick` handler is provider?

Reply:
Yes, you are right. I will remove this and the div example. Thanks!